src/System.Private.CoreLib/shared/System/String.Searching.cs

```diff
@@ -260,10 +260,10 @@ public int IndexOf(string value, int startIndex, int count, StringComparison com
             if (value == null)
                 throw new ArgumentNullException(nameof(value));
 
-            if (startIndex < 0 || startIndex > this.Length)
+            if ((uint)startIndex > (uint)this.Length)
                 throw new ArgumentOutOfRangeException(nameof(startIndex), SR.ArgumentOutOfRange_Index);
 
-            if (count < 0 || startIndex > this.Length - count)
+            if ((uint)startIndex > (uint)(this.Length - count))
```

Comment:
This optimization isn't valid. Still need to check _count_ for a negative value.